src/Chess/Game/ChessGame.php

```diff
@@ -1293,15 +1296,16 @@ public function inStaleMate($color = null)
             if (count($canMove)) {
                 $a = $this->getPiece($name);
                 foreach ($canMove as $move) {
-                    $this->startTransaction();
                     $this->_move = $color;
 
                     if (!$this->isError($this->moveSquare($a, $move))) {
+                        $this->rollbackTransaction();
```

Comment:
I think this is the only thing that was missing, but the others add no real overhead and so why not? :)